internal/advertisement-operator/config-watcher.go

```diff
@@ -4,91 +4,65 @@ import (
 	"context"
 	protocolv1 "github.com/liqoTech/liqo/api/advertisement-operator/v1"
 	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
+	"github.com/liqoTech/liqo/pkg/clusterConfig"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/klog"
 )
 
-func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string) error {
-	configClient, err := policyv1.CreateClusterConfigClient(kubeconfigPath)
-	if err != nil {
-		klog.Error(err)
-		return err
-	}
-
-	watcher, err := configClient.Resource("clusterconfigs").Watch(metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-
-	go func() {
-		for event := range watcher.ResultChan() {
-			configuration, ok := event.Object.(*policyv1.ClusterConfig)
-			if !ok {
-				continue
-			}
+func (b *AdvertisementBroadcaster) WatchConfiguration() {
```

Comment:
@fraborg Do you need to specify kubeconfigPath? Or trying InCluster and, if failed, on ~/.kube/config is enough?

Reply:
I have reused `NewKubeconfig()` function (https://github.com/LiqoTech/liqo/blob/master/pkg/crdClient/v1alpha1/client.go#L29) which receives the kubeconfigPath.
I think it's useful because with that you can pass whichever kubeconfig you want, it not necessarily needs to be located in `.kube/config`

Reply:
ok, so I add this parameter, if not set it uses default path